cache/edge_lambdas/src/redirector.ts

```diff
@@ -70,24 +70,39 @@ export const getRedirect = (
     return redirect301(host, literalRedirects[uriSansSlash]);
   }
 
-  if (request.querystring && queryRedirects[uriSansSlash]) {
-    const potentialRedirect = queryRedirects[uriSansSlash];
+  if (queryRedirects[uriSansSlash]) {
     const requestParams = new URLSearchParams(request.querystring);
-    // A redirect occurs if all of the params in the redirect rule are contained
-    // within the request
-    if (paramsAreSubset(requestParams, potentialRedirect.matchParams)) {
-      // Only forward params that are in `forwardParams`
-      const newParams = filterParams(
-        requestParams,
-        potentialRedirect.forwardParams
-      );
-      const requestParamsString = newParams.toString();
-      return redirect301(
-        host,
-        requestParamsString
-          ? potentialRedirect.redirectPath + '?' + requestParamsString
-          : potentialRedirect.redirectPath
-      );
+
+    // If the redirect has matchParams, pick the relevant one from the list
+    // Otherwise return the one that has none
+    const potentialRedirect = queryRedirects[uriSansSlash].find(q =>
```

Comment:
What if multiple redirects match? Maybe we should `filter` (not `find`) and error if the resultant array has more than 1 element?

Reply:
Happy to change, and to make it an error, because how would more than one match? Let's hope we never need that 😄